clang/lib/CodeGen/Targets/SPIR.cpp

```diff
@@ -240,6 +243,31 @@ void CommonSPIRTargetCodeGenInfo::setOCLKernelStubCallingConvention(
       FT, FT->getExtInfo().withCallingConv(CC_SpirFunction));
 }
 
+// LLVM currently assumes a null pointer has the bit pattern 0, but some GPU
+// targets use a non-zero encoding for null in certain address spaces.
+// Because SPIR(-V) is a virtual target and the bit pattern of a non-generic
+// null is unspecified, materialize non-generic null via an addrspacecast from
+// the generic null.
+// This allows later lowering to substitute the target’s real sentinel value.
+llvm::Constant *
+CommonSPIRTargetCodeGenInfo::getNullPointer(const CodeGen::CodeGenModule &CGM,
+                                            llvm::PointerType *PT,
+                                            QualType QT) const {
+  LangAS AS;
+  if (QT->getUnqualifiedDesugaredType()->isNullPtrType())
+    AS = LangAS::Default;
+  else
+    AS = QT->getPointeeType().getAddressSpace();
```

Comment:
Initialize with ?: instead of conditional assign

Reply:
> Initialize with ?: instead of conditional assign

done in https://github.com/llvm/llvm-project/pull/161773/commits/b95e77aaea2579f83e209fa66058a989a6c93fc0